sawtooth/sample-sawtooth-java/src/main/java/InventoryClientSender.java

```diff
@@ -30,18 +36,43 @@
 	private static final Logger logger = Logger.getLogger(InventoryClientSender.class.getName());
 	private static final String IDEM = "invent";
 	private static final String VER = "1.0";
-	private static final String ITEM_ID = "00002";
-	public static void main(String[] args) throws UnirestException, UnsupportedEncodingException, InternalError {
+	private static final String ITEM_ID = "00006";
+	public static void main(String[] args) throws UnirestException, UnsupportedEncodingException, InternalError, ParseException {
+		//putDataToBLockchain();
+		getDataFromBlockchain();		
+	}
+	
+	public static void getDataFromBlockchain() throws UnirestException, UnsupportedEncodingException, InternalError, ParseException {
 
+		String address = getAddress(IDEM, ITEM_ID); // get unique address for input and output
+		
+		String serverResponse = Unirest.get("http://localhost:8008/state")
+				.queryString("address",address)
+				.asString()
+				.getBody();
+		
+		System.out.println(serverResponse);
+		
+		StateData.Builder stateBuilder = StateData.newBuilder();
+		JsonFormat.merge(serverResponse, stateBuilder);
+		StateData stateData = stateBuilder.build();
+		Decoder decoder = Base64.getDecoder();
+		for(Data data : stateData.getDataList()) {
+			byte[] valueDecoded = decoder.decode(data.getData());
```

Comment:
These bytes can be read in to a protobuf object which we can define. Lets take it as lower priority.